TdmPrototypeBackend.Types/Movement.cs

```diff
@@ -114,22 +111,16 @@ public List<int> _MatchReferences
         set => matchReferences = value;
     }
 
-    public void AddRelationship(string type, TdmRelationshipObject relationship)
+    public void ClearRelationships()
     {
-        Relationships.Notifications.Links ??= relationship.Links;
-        foreach (var dataItem in relationship.Data)
-        {
-            if (Relationships.Notifications.Data.All(x => x.Id != dataItem.Id))
-            {
-                Relationships.Notifications.Data.Add(dataItem);
-            }
-        }
+        Relationships.Notifications.Data.Clear();
+    }
 
-        var allDocsMatched = Items
-            .Select(x => x.ItemNumber)
-            .All(itemNumber => Relationships.Notifications.Data.Any(x => x.Matched && x.SourceItem == itemNumber));
-        
-        Relationships.Notifications.Matched = relationship.Matched;
+    public void AddRelationship(TdmRelationshipObject relationship)
+    {
+        Relationships.Notifications.Links ??= relationship.Links;
+        Relationships.Notifications.Data.AddRange(relationship.Data);
+        Relationships.Notifications.Matched = Relationships.Notifications.Data.All(x => x.Matched);
```

Comment:
I am not sure this is quite right, lets have a chat around it, incase I am mis understanding